README.md

```diff
@@ -6,39 +6,157 @@ Please edit this README.md file with your answers to the worksheet questions.
+### |FALSE|FALSE|FALSE|    FALSE   |  FALSE  |          FALSE          |
+### |FALSE|FALSE|TRUE |    FALSE   |  FALSE  |          FALSE          |
+### |FALSE|TRUE |FALSE|    FALSE   |  FALSE  |          FALSE          |
+### |FALSE|TRUE |TRUE |    FALSE   |  FALSE  |          FALSE          |
+### |TRUE |FALSE|FALSE|    FALSE   |  FALSE  |          FALSE          |
+### |TRUE |FALSE|TRUE |    FALSE   |  TRUE   |          TRUE           |
+### |TRUE |TRUE |FALSE|    TRUE    |  FALSE  |          TRUE           |
+### |TRUE |TRUE |TRUE |    TRUE    |  TRUE   |          TRUE           |
+
+### |  A  |  B  |  C  |            | B or C |       A and (B or C)     |
+### |FALSE|FALSE|FALSE|            |  FALSE |           FALSE          |
+### |FALSE|FALSE|TRUE |            |  TRUE  |           FALSE          |
+### |FALSE|TRUE |FALSE|            |  TRUE  |           FALSE          |
+### |FALSE|TRUE |TRUE |            |  TRUE  |           FALSE          |
+### |TRUE |FALSE|FALSE|            |  FALSE |           FALSE          |
+### |TRUE |FALSE|TRUE |            |  TRUE  |           TRUE           |
+### |TRUE |TRUE |FALSE|            |  TRUE  |           TRUE           |
+### |TRUE |TRUE |TRUE |            |  TRUE  |           TRUE           |
+
 ### d
 
+### |  A  |  B  |  C  | A or B | A or C | (A or B) & (A or C) |
+### |FALSE|FALSE|FALSE|  FALSE |  FALSE |        FALSE        |
+### |FALSE|FALSE|TRUE |  FALSE |  TRUE  |        FALSE        |
+### |FALSE|TRUE |FALSE|  TRUE  |  FALSE |        FALSE        |
+### |FALSE|TRUE |TRUE |  TRUE  |  TRUE  |        TRUE         |
+### |TRUE |FALSE|FALSE|  TRUE  |  TRUE  |        TRUE         |
+### |TRUE |FALSE|TRUE |  TRUE  |  TRUE  |        TRUE         |
+### |TRUE |TRUE |FALSE|  TRUE  |  TRUE  |        TRUE         |
+### |TRUE |TRUE |TRUE |  TRUE  |  TRUE  |        TRUE         |
+
+### |  A  |  B  |  C  |        | B & C |      A or (B & C)    |
+### |FALSE|FALSE|FALSE|        | FALSE |         FALSE        |
+### |FALSE|FALSE|TRUE |        | FALSE |         FALSE        |
+### |FALSE|TRUE |FALSE|        | FALSE |         FALSE        |
+### |FALSE|TRUE |TRUE |        | TRUE  |         TRUE         |
+### |TRUE |FALSE|FALSE|        | FALSE |         TRUE         |
+### |TRUE |FALSE|TRUE |        | FALSE |         TRUE         |
+### |TRUE |TRUE |FALSE|        | FALSE |         TRUE         |
+### |TRUE |TRUE |TRUE |        | TRUE  |         TRUE         |
+
 ## Question 4
 
 ### a
 
+### The two programs are equivalent because they follow the same logic as the truth table that I presented in Question 3a. The first code listing is printing a text line if theres no a.txt and no b.txt which is another way of saying NOT A AND NOT B. The second code listing is printing a text line if a.txt or b.txt doesn't exist which is another way of putting NOT(A OR B). Applying these to a truth table as like I did before, you can see the same outputs for both.
```

Comment:
Well explained -- you're on the right track to apply this kind of reasoning to the other parts of Q4